### PR TITLE
XML data host and file reference variables

### PR DESCRIPTION
**Pull request by ilidio-lopes**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Check the test cases in the PR . XML data host variables can be in a sample code

```cobol
        Identification Division.
        Program-Id. 'TEST1'.
        Data Division.
         Working-Storage Section.
       01 VAR3 USAGE IS SQL TYPE IS XML AS CHARACTER LARGE OBJECT (10 M).
       01 VAR-NAME4 USAGE IS SQL TYPE IS XML AS CHAR LARGE OBJECT (10 G).
       01  VAR-NAME5 USAGE IS SQL TYPE IS XML AS CLOB (20)
       01  VAR-NAME6 USAGE IS SQL TYPE IS XML AS DBCLOB (30 K)
       01  VAR-NAME7 USAGE IS SQL TYPE IS XML AS BLOB-LOCATOR.
       01  VAR-NAME8 USAGE IS SQL TYPE IS XML AS CLOB-LOCATOR.
        PROCEDURE division
```

image

- [ ] Test A
- [ ] Test B

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by Nurkambay** [comment]
@ilidio-lopes please resolve conflicts

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java [hunk 1]
```diff
@@ -94,6 +94,27 @@ public class TestSqlHostVariable {
...
+                  + "       01 {$*VAR3} USAGE IS SQL TYPE IS XML AS CHARACTER LARGE OBJECT (10 M).\n"
+                  + "       01 {$*VAR-NAME4} USAGE IS SQL TYPE IS XML AS CHAR LARGE OBJECT (10 G).\n"
+                  + "       01 {$*VAR-NAME5} USAGE IS SQL TYPE IS XML AS CLOB (20).\n"
+                  + "       01 {$*VAR-NAME6} USAGE IS SQL TYPE IS XML AS DBCLOB (30 K).\n"
+                  + "       01 {$*VAR-NAME7} USAGE IS SQL TYPE IS XML AS BLOB-LOCATOR.\n"
+                  + "       01 {$*VAR-NAME8} USAGE IS SQL TYPE IS XML AS CLOB-LOCATOR.\n"
+                  + "       01 {$*VAR-NAME9} USAGE IS SQL TYPE IS XML AS DBCLOB-LOCATOR.\n"
+                  + "       01 {$*VAR-NAME10} USAGE IS SQL TYPE IS XML AS  BLOB-FILE.\n"
+                  + "       01 {$*VAR-NAME11} USAGE IS SQL TYPE IS XML AS  CLOB-FILE.\n"
+                  + "       01 {$*VAR-NAME12} USAGE IS SQL TYPE IS XML AS  DBCLOB-FILE.\n"
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:114]: Can we add a use case for level 2-48 and a level 49 with error

**Comment by amanPras** [comment]
I suggest rebase with development. Can see 2 unnecessary commits  bdc108051dce63bf1233ff2d639b6ef2d4e17a2c and 5890e351d3c6c7fd114d8f90c0ec63558af61dd0
rest LGTM

**Review by amanPras — CHANGES_REQUESTED** [review]
1. add a use case for level 2-48 and a level 49 with error
2. Two unnecessary commits https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/commit/bdc108051dce63bf1233ff2d639b6ef2d4e17a2c and https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/commit/5890e351d3c6c7fd114d8f90c0ec63558af61dd0

File: server/engine/src/test/java/org/eclipse/lsp/cobol/lsp/SourceUnitGraphTest.java [hunk 2]
```diff
@@ -103,10 +103,10 @@ void testLinksBetweenCopybookAndSourceCode() {
...
 
     assertEquals("COPY 3 TEXT", sourceUnitGraph.getCopyNodeContent(copyNode3));
-    assertTrue(sourceUnitGraph.isCopybook(copy1Uri));
-    assertTrue(sourceUnitGraph.isCopybook(copy2Uri));
-    assertTrue(sourceUnitGraph.isCopybook(copy3Uri));
-    assertFalse(sourceUnitGraph.isCopybook(URI));
+    assertTrue(sourceUnitGraph.isUserSuppliedCopybook(copy1Uri));
+    assertTrue(sourceUnitGraph.isUserSuppliedCopybook(copy2Uri));
+    assertTrue(sourceUnitGraph.isUserSuppliedCopybook(copy3Uri));
+    assertFalse(sourceUnitGraph.isUserSuppliedCopybook(URI));
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/lsp/SourceUnitGraphTest.java]: These changes looks like are from 8bd8a7135735dd185a3a0a6aebd042b6e19260fb , and unrelate to this PR

**Review by amanPras — APPROVED** [review]

**Pull request merged**